flink-end-to-end-tests/flink-kafka-test/pom.xml

```diff
@@ -0,0 +1,91 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+		 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+		 xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<parent>
+		<artifactId>flink-end-to-end-tests</artifactId>
+		<groupId>org.apache.flink</groupId>
+		<version>1.7-SNAPSHOT</version>
+	</parent>
+	<modelVersion>4.0.0</modelVersion>
+
+	<artifactId>flink-kafka-test</artifactId>
+	<name>flink-kafka-test</name>
+
+	<dependencies>
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-kafka-test-base</artifactId>
+			<version>${project.version}</version>
+		</dependency>
+
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-connector-kafka-base_${scala.binary.version}</artifactId>
```

Comment:
I didn't introduce it at first, this was done to explicitly exclude the base module's dependency on the lower version of the kafka client. Otherwise it will cause a compilation error.

Reply:
What was the error? We should definitely look into it, since we shouldn't expect users to use the same hack in their poms.

Reply:
```
03:59:30.102 [WARNING] 
Dependency convergence error for org.apache.kafka:kafka-clients:0.10.2.1 paths to dependency are:
+-org.apache.flink:flink-kafka-test:1.7-SNAPSHOT
  +-org.apache.flink:flink-connector-kafka_2.11:1.7-SNAPSHOT
    +-org.apache.flink:flink-connector-kafka-base_2.11:1.7-SNAPSHOT
      +-org.apache.kafka:kafka-clients:0.10.2.1
and
+-org.apache.flink:flink-kafka-test:1.7-SNAPSHOT
  +-org.apache.flink:flink-connector-kafka_2.11:1.7-SNAPSHOT
    +-org.apache.kafka:kafka-clients:2.0.0

03:59:30.103 [WARNING] Rule 0: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
Failed while enforcing releasability. See above detailed error message.
```

details : https://api.travis-ci.org/v3/job/449173244/log.txt

Reply:
OK, agree.